File: problem10/problem10.py
def prime(n):
	from math import sqrt, floor
	if n <= 1:
		return False
	for i in xrange(2, int(floor(sqrt(n)))+1):
		if n%i ==0:
			return False
	return True

def sum_primes():
	return sum([i for i in xrange(2,2000000) if prime(i)])

print sum_primes()

File: problem10/problem10.R
prime <- function(n) {
	if(n <= 1) {
		return(FALSE)
	}
	if(n == 2) {
		return(TRUE)
	}
	for(i in 1:floor(sqrt(n))+1) {
		if(n%%i == 0) {
			return(FALSE)
		}
	}
	return(TRUE)
}

sum_primes <- function() {
	l <- 0
	for(i in 1:2000000) {
		if(prime(i)) {
			l <- l + i
		}
	}
	l
}

sum_primes()

